js/source/esri-tile-pyramid.js

```diff
@@ -0,0 +1,321 @@
+'use strict';
+
+var Tile = require('mapbox-gl/js/source/tile');
+var TileCoord = require('mapbox-gl/js/source/tile_coord');
+var Point = require('mapbox-gl/node_modules/point-geometry');
+var Cache = require('mapbox-gl/js/util/mru_cache');
+var util = require('mapbox-gl/js/util/util');
+
+module.exports = TilePyramid;
+
+function TilePyramid(options) {
```

Comment:
I've removed the concept of "indexedVector" for now, and all new files (esri-*) and datasources that were originally in this branch/PR are gone. In its place I've modified 5 files (in each very subtly). Will call each change out separately below.